types/rlp_marshal.go

```diff
@@ -91,6 +91,8 @@ func (h *Header) MarshalRLPWith(arena *fastrlp.Arena) *fastrlp.Value {
 	vv.Set(arena.NewBytes(h.MixHash.Bytes()))
 	vv.Set(arena.NewCopyBytes(h.Nonce[:]))
 
+	vv.Set(arena.NewUint(h.BaseFee))
```

Comment:
How is this backward compatible? Chains running the old version will not have this field.

Reply:
We should discuss this point a bit more because it seems important the backward compatibility. What geth does is that this field is a point and it is only set (marshalled) if `London` is enabled. I do not have a clear answer to this right now.

Reply:
Added TODO before this line so we could discuss it later.